### ISSUE TITLE
Support custom builder in cordova-serve

### ISSUE THREAD

**Issue by doug-a-brunner** [ionic-team/angular-toolkit#108]
In the resolution of issue #32 the capability was added to use browserTarget to specify an arbitrary builder that the `cordova-build` builder wraps. It would be very helpful to have the analogous capability for `cordova-serve`: rather than hardcoding `DevServerBuilder` and extending it, use `getBuilder` to pick up a specified builder.

It looks, however, like this might not be possible with Angular Architect as it stands now: their `DevServerBuilder` seems to be grabbing the browser build options out of the specified `browserTarget`, and doing so internally within `run()`. That would make an override of `buildWebpackConfig()` the only place where the browser build options are accessible to modify...and this is not part of the `Builder<>` interface, only of `DevServerBuilder`, so this strategy doesn't work for an arbitrary Builder loaded by Architect.

Does this seem right - and if so, is there any resolution coming with the new Angular build API? I haven't been able to find any documentation on the new API, so hard to tell what might be changing with it. This problem throws a bit of a wrench into my development, because my project (which builds for both Cordova and Electron with a single codebase) relies on a custom webpack config to ignore imports of node-provided modules like `fs` when building for Angular.

**Comment by bobrosoft** [comment]
I think I have related issue. Just added support for custom webpack configuration using that article https://www.techiediaries.com/ionic-angular-typeorm-custom-webpack-configuration/   and have it working for `ionic serve` but not with `ionic serve --devapp` because second is using ionic-cordova-serve architect builder which looks as not picking up next correct builder in the chain but ionic-cordova-build does it, looks as.

**Comment by mhartington** [comment]
Hi there! Is this still an issue?With the latest updates to the new builder API, this should be supported as the browserBuild is an option that can be passed

```
          "options": {
            "cordovaBuildTarget": "app:ionic-cordova-build",
            "devServerTarget": "app:serve"
          },
```

**Comment by bobrosoft** [comment]
@mhartington working now! Good job 👍👍👍

**Comment by doug-a-brunner** [comment]
@mhartington, I don't fully understand your comment. When you say `browserBuild`, where does that get specified and what sort of values does it take? I grepped through a freshly created Ionic example app, and only saw `browserBuild` used as a variable name in `angular-toolkit`, not in any schema or config file. (A tabs app created from the CLI right now has a schema path in its `angular.json` that no longer exists, making it hard to follow a trail.) I haven't been able to find any documentation on this so far - is there some that I'm missing?

**Comment by osben** [comment]
please add support custom-webpack

**Comment by lucashsilva** [comment]
I am facing the same issue. My webpack configuration from build task is ignored by ionic-cordova-serve task.

**Comment by freedom-git** [comment]
Same issue, is there a solution？

**Comment by mhartington** [comment]
Going to close this. As I mentioned here https://github.com/ionic-team/angular-toolkit/issues/230#issuecomment-728929929

We need to be explicit about what kind of builder we trigger.

**Comment by Bishop19** [comment]
HI! Any updates on this? I'm really struggling on this 😞
